### PR TITLE
Adds Uniformly controlled rotations (Ry and Rz)

### PR DESCRIPTION
**Pull request by damiansteiger**

### CODE REVIEW

**Comment by damiansteiger** [comment]
Thanks for the suggestions. They have been implemented.

File: projectq/ops/_uniformly_controlled_rotation.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        .. code-block:: python
+
+        controls = eng.allocate_qureg(2)
+        target = eng.allocate_qubit()
+        UniformlyControlledRy(angles=[0.1, 0.2, 0.3, 0.4]) | (controls, target)
+
+    Note:
+        The first quantum register contains the control qubits. When converting
+        the classical state k of the control qubits to an integer, we define
+        controls[0] to be the least significant (qu)bit.
```
Comment (damiansteiger) [inline projectq/ops/_uniformly_controlled_rotation.py]: On second thoughts:
Which order is more natural?

Comment (thomashaener) [reply projectq/ops/_uniformly_controlled_rotation.py]: Since it's about integers, I'd also choose this ordering of bits.

**Comment by damiansteiger** [comment]
Something is not yet working as expected:
```
import projectq
from projectq.setups import restrictedgateset
from projectq.ops import Ry, Rz, UniformlyControlledRy, UniformlyControlledRz

engine_list = restrictedgateset.get_engine_list(one_qubit_gates=(Ry,Rz))
eng = projectq.MainEngine(engine_list=engine_list, verbose=True)

qureg = eng.allocate_qureg(4)
eng.flush()

UniformlyControlledRz([0,1]) | ([qureg[3]], qureg[2])
eng.flush() # just outcommenting this triggers the error... :-(
UniformlyControlledRz([0.5, 0.8, 1.2, 2.5, 4.4, 2.32, 6.6, 15.12]) | (qureg[1:], qureg[0])
eng.flush()
```

**Comment by thomashaener** [comment]
Okay, fixed after some interesting discussion :-)

**Comment by damiansteiger** [comment]
🙈
Can I go ahead with this?

**Comment by damiansteiger** [comment]
One more addition is coming asap

**Pull request merged**